tasks/ThrustersInput.cpp

```diff
@@ -0,0 +1,192 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "ThrustersInput.hpp"
+
+using namespace auv_control;
+
+ThrustersInput::ThrustersInput(std::string const& name)
+    : ThrustersInputBase(name)
+{
+}
+
+ThrustersInput::ThrustersInput(std::string const& name, RTT::ExecutionEngine* engine)
+    : ThrustersInputBase(name, engine)
+{
+}
+
+ThrustersInput::~ThrustersInput()
+{
+}
+
+
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See ThrustersInput.hpp for more detailed
+// documentation about them.
+
+bool ThrustersInput::configureHook()
+{
+    if (! ThrustersInputBase::configureHook())
+        return false;
+
+    thrusterVoltage		= _thruster_voltage.get();
+    coeffPos 			= _thruster_coeff_pos.get();
+    coeffNeg 			= _thruster_coeff_neg.get();
+    controlModes		= _control_modes.get();
+
+    // Let's assume control_modes was set with the right number of thrusters
+    numberOfThrusters = controlModes.size();
+
+    // Those coefficients need to be set
+    if(coeffPos.empty() || coeffNeg.empty())
+    {
+		RTT::log(RTT::Error) << "Thruster coefficients were not set" << RTT::endlog();
+    	exception(WRONG_SIZE_OF_THRUSTER_COEFFICIENTS);
+    }
+
+
+    // Check size of parameters. They should met among them.
+    if(coeffPos.size() != numberOfThrusters || coeffNeg.size() != numberOfThrusters )
+    {
+    	if(coeffPos.size() != coeffNeg.size())
```

Comment:
This makes the whole logic hard to read, and I believe will be hard to understand on the usage side as well. All three have to agree, so test for that and output a generic exception (THRUSTER_ARRAYS_SIZE_DIFFER), message

```
The size of control_modes, thruster_coeff_pos and thruster_coeff_neg do not agree (... output the size of the 3 ...)
```